recipes-bsp/u-boot/files/0008-board-siemens-iot2050-Pass-DDR-size-from-FSBL.patch

```diff
@@ -0,0 +1,109 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Baocheng Su <baocheng.su@siemens.com>
+Date: Wed, 29 Nov 2023 23:16:15 +0800
+Subject: [PATCH] board: siemens: iot2050: Pass DDR size from FSBL
+
+Due to new DDR size introduction, the current logic of determining the
+DDR size is not able to get the correct size.
+
+Instead, the DDR size is determined by the FSBL(SEBOOT) then passed to
+u-boot through the scratchpad info.
+
+The SEBoot version must be >= D/V01.04.0x.0x to support this change.
+
+Also now for some variants, the DDR size may > 2GB, so borrow some code
+from the TI evm to iot2050 to support more than 2GB DDR.
+
+Signed-off-by: Baocheng Su <baocheng.su@siemens.com>
+---
+ board/siemens/iot2050/board.c | 45 ++++++++++++++++++++++++++++-------
+ include/configs/iot2050.h     |  3 +++
+ 2 files changed, 39 insertions(+), 9 deletions(-)
+
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index b83d5f8669e87db05935e826e431a647b5056987..ca1e1336bbd7ecf45d52cf2cda7ca576f1ce4925 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -38,6 +38,8 @@ struct iot2050_info {
+ 	u8 mac_addr_cnt;
+ 	u8 mac_addr[8][ARP_HLEN];
+ 	char seboot_version[40 + 1];
++	u8 padding[3];
++	u32 ddr_size_mb;
+ } __packed;
+ 
+ /*
+@@ -341,25 +343,50 @@ int board_init(void)
+ 
+ int dram_init(void)
+ {
+-	if (board_is_advanced())
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++#ifdef CONFIG_PHYS_64BIT
++	gd->ram_size = ((phys_size_t)(info->ddr_size_mb)) << 20;
++#else
++	if (info->ddr_size_mb > 2048)
+ 		gd->ram_size = SZ_2G;
+ 	else
+-		gd->ram_size = SZ_1G;
+-
++		gd->ram_size = ((phys_size_t)(info->ddr_size_mb)) << 20;
++#endif
+ 	return 0;
+ }
+ 
++ulong board_get_usable_ram_top(ulong total_size)
++{
++#ifdef CONFIG_PHYS_64BIT
```

Comment:
Isn't that always true for the IOT2050?

Reply:
We only use u-boot for a53, so this is always true for us. I was thinking to be compatible with the R5 building, however, we don't provide R5 dts at all, this seems a fake compatibility.

Acked and will removed.

Reply:
Fixed in #509